lib/uknetdev/include/uk/netbuf.h

```diff
@@ -101,10 +101,34 @@ typedef void (*uk_netbuf_dtor_t)(struct uk_netbuf *);
  * uk_netbuf_prepare_buf() are placing all these three regions into a single
  * allocation.
  */
+
+/* Packet is validated (non-corrupted, checksum okay)
+ * When sending or receiving a packet with a partial checksum, the
+ * DATA_VALID bit should also be set in order to confirm that the
+ * payload is not corrupted.
+ */
+#define UK_NETBUF_F_DATA_VALID_BIT   0
+#define UK_NETBUF_F_DATA_VALID       (1 << UK_NETBUF_F_DATA_VALID_BIT)
+
+/* Indicates that a protocol Checksum was only partially computed (e.g, TCP,
+ * UDP). `csum_start` and `csum_offset` fields indicate over which bytes
+ * the checksum needs to be computed and to which offset the result
+ * needs to be written. On transmit this is typically done by a physical
+ * interface card when leaving the host.
+ * NOTE: The defacto standard is that IP header checksums are always
+ * computed in software). In the case of TCP and UDP, a softrware computed
+ * partial checksum (just the pdeudo header) is filled into the checksum
+ * field.
+ */
+#define UK_NETBUF_F_PARTIAL_CSUM_BIT 1
+#define UK_NETBUF_F_PARTIAL_CSUM     (1 << UK_NETBUF_F_PARTIAL_CSUM_BIT)
+
 struct uk_netbuf {
 	struct uk_netbuf *next;
 	struct uk_netbuf *prev;
 
+	uint8_t flags;         /**< Flags for this netbuf */
```

Comment:
Since this struct is never used to interact with an actual device directly, padding doesn't matter. A driver will parse the fields and will feed device specific descriptors. Whatever is making accessing fields in this struct faster is actually good for the networking subsystem - so I prefer leaving it up to the compiler. I don't expect many flags in the future but we can still extend its data type to `uint16_t` when needed since everything is recompiled together. The `uint8_t` was followed the `pbuf` example in lwip.

Reply:
Sounds good. I just wanted to highlight that we have space for more if needed 😉